### DISCUSSION TITLE
ReactPy Performance?

### DISCUSSION THREAD

**Discussion by broshaun** [idom-team/idom#1092] [Question] [answered]
I can finally get rid of JS and use my favorite Python. But all rendering is sent on the server side, so is there any problem with efficiency? The reason why everyone uses JS is because the operation is carried out on the browser. If we can move most of the functions of rendering and computing to wasm, will it solve the problem of server-side rendering burden?

**Comment by Archmonger** [comment, marked as answer]
Client-side components are planned to be supported in the future (#110).

However, server-side performance is generally not a problem for small/mid scale projects. A lot of our server-side backends are [impressively fast](https://klen.github.io/py-frameworks-bench/) even when only running with one [ASGI worker](https://www.uvicorn.org/settings/#production) (~80 MB of RAM). If you're servicing more users than 100 ASGI workers (8 GB of RAM) can handle, we realistically talking about scales of massive websites. 

You really should only focus on optimizations like this when reaching the scales of Google/YouTube/Facebook. Even then, some of the biggest websites in the world use relatively slow server-side frameworks (such as YouTube using Django). Even with sites like YouTube, server side rendering is not an impossible hurdle due to distributed computing and caching.
